test/semantics/call13.f90

```diff
@@ -0,0 +1,49 @@
+! Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+!
+! Licensed under the Apache License, Version 2.0 (the "License");
+! you may not use this file except in compliance with the License.
+! You may obtain a copy of the License at
+!
+!     http://www.apache.org/licenses/LICENSE-2.0
+!
+! Unless required by applicable law or agreed to in writing, software
+! distributed under the License is distributed on an "AS IS" BASIS,
+! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+! See the License for the specific language governing permissions and
+! limitations under the License.
+
+! Test 15.4.2.2 constraints and restrictions for calls to implicit
+! interfaces
+
+subroutine s(assumedRank, coarray, class, classStar, typeStar)
+  type :: t
+  end type
+
+  real :: assumedRank(..), coarray[*]
+  class(t) :: class
+  class(*) :: classStar
+  type(*) :: typeStar
+
+  type :: pdt(len)
+    integer, len :: len
+  end type
+  type(pdt(1)) :: pdtx
+
+  !ERROR: The expression (int(implicit01(),kind=8)) cannot be used as a specification expression (reference to impure function 'implicit01')
+  real :: array(implicit01())  ! 15.4.2.2(2)
```

Comment:
I'm not sure it's worth including the expression in the error message: it might be long and (as in this case) it might not look like the original expression. As long as the source location is pointing at the expression, I think a message like this is better:
`Invalid specification expression: reference to impure function 'implicit01'`

Reply:
Will remove the expression.